File: nahomy-c2-backend/src/domain/entities/log_auditoria.py
from datetime import datetime


class LogAuditoria:
    def __init__(
        self,
        id: int,
        usuario_id: str | None,
        proceso_id: int | None,
        accion: str,
        tabla_afectada: str,
        detalles: str | None,
        fecha_registro: datetime,
    ):
        self._id = id
        self._usuario_id = usuario_id
        self._proceso_id = proceso_id
        self._accion = accion
        self._tabla_afectada = tabla_afectada
        self._detalles = detalles
        self._fecha_registro = fecha_registro

    @property
    def id(self) -> int:
        return self._id

    @property
    def usuario_id(self) -> str | None:
        return self._usuario_id

    @property
    def proceso_id(self) -> int | None:
        return self._proceso_id

    @property
    def accion(self) -> str:
        return self._accion

    @property
    def tabla_afectada(self) -> str:
        return self._tabla_afectada

    @property
    def detalles(self) -> str | None:
        return self._detalles

    @property
    def fecha_registro(self) -> datetime:
        return self._fecha_registro

File: nahomy-c2-backend/src/domain/entities/LogAuditoria.ts
export class LogAuditoria {
  constructor(
    private readonly _id: number,
    private readonly _usuarioId: string | null,
    private readonly _procesoId: number | null,
    private readonly _accion: string,
    private readonly _tablaAfectada: string,
    private readonly _detalles: string | null,
    private readonly _fechaRegistro: Date,
  ) {}

  get id(): number {
    return this._id
  }

  get usuarioId(): string | null {
    return this._usuarioId
  }

  get procesoId(): number | null {
    return this._procesoId
  }

  get accion(): string {
    return this._accion
  }

  get tablaAfectada(): string {
    return this._tablaAfectada
  }

  get detalles(): string | null {
    return this._detalles
  }

  get fechaRegistro(): Date {
    return this._fechaRegistro
  }
}
